cv_next/server/api/users.ts

```diff
@@ -382,3 +375,33 @@ export async function setDisplayName(
     });
   }
 }
+
+/*
+ * Set the first login status of the current user.
+ *
+ * @param {boolean} isFirstLogin - first login status.
+ * @return {Promise<Result<void, string>>} A promise that resolves with void or rejects with an error message.
+ */
+export async function setFirstLogin(
+  isFirstLogin: boolean
+): Promise<Result<void, string>> {
+  const id = await getCurrentId();
+  if (!id.ok || !id.val) {
+    return Err(setFirstLogin.name, { err: Error("No user was connected!") });
+  }
+
+  try {
+    const { error } =
+      await SupabaseHelper.getSupabaseInstance().auth.updateUser({
+        data: { is_first_login: isFirstLogin },
```

Comment:
Not entirely sure of the difference, perhaps @Aradon267 knows, but why `updateUser` and not regular table update (like the other functions in this file)?

Reply:
UpdateUser updates the authentication metadata from the 3rd party provider(auth.users) unlike the other that updates our data(public.profiles)

Reply:
Is it needed here? 🤔

Reply:
yes, that is what we agreed in #201.
This is meant to be the setup for the `first_login` state